libheif/heif_colorconversion.cc

```diff
@@ -2885,13 +2885,14 @@ bool ColorConversionPipeline::construct_pipeline(ColorState input_state,
 
     // expand the node with minimum cost
 
-    for (size_t i = 0; i < ops.size(); i++) {
+    for (const auto& op_ptr : ops) {
 
 #if DEBUG_PIPELINE_CREATION
-      std::cerr << "-- apply op: " << typeid(*(ops[i])).name() << "\n";
+      auto& op = *op_ptr;
+      std::cerr << "-- apply op: " << typeid(op).name() << "\n";
```

Comment:
This is the first of the two warning fixes in this CL. This warning is emitted only when DEBUG_PIPELINE_CREATION is defined as 1.